default/python/universe_generation/monsters.py

```diff
@@ -1,8 +1,110 @@
+import sys
 import random
 import freeorion as fo
 import util
 import statistics
 import universe_tables
+from galaxy import DisjointSets
+
+
+def populate_monster_fleet(fleet_plan, system):
+    """
+    Create a monster fleet in ''system'' according to ''fleet_plan''
+    """
+
+    # create monster fleet
+    monster_fleet = fo.create_monster_fleet(system)
+    # if fleet creation fails, report an error and try to continue with next system
+    if monster_fleet == fo.invalid_object():
+        errstr = "Python generate_monsters: unable to create new monster fleet %s" % fleet_plan.name()
+        util.report_error(errstr)
+        raise RuntimeError(errstr)
+
+    # add monsters to fleet
+    for design in fleet_plan.ship_designs():
+        # create monster, if creation fails, report an error and try to continue with the next design
+        if fo.create_monster(design, monster_fleet) == fo.invalid_object():
+            errstr = "Python generate_monsters: unable to create monster %s" % design
+            util.report_error(errstr)
+            raise RuntimeError(errstr)
+
+    print "Spawn", fleet_plan.name(), "at", fo.get_name(system)
+
+monsters_that_alter_starlanes = {"SM_EXP_OUTPOST"}
+
+
+def place_experimentors(systems):
+    """
+    Place the expermentor outpost system.  Check that removing all of the
+    starlanes entering the system does not break the rest of the galaxy
+    into more than one connected piece.
+    """
+
+    random.shuffle(systems)
+
+    # Find the Experimentor output fleet plan
+    plans = [plan for plan in fo.load_monster_fleet_plan_list()
+             if (plan.name() in monsters_that_alter_starlanes
+                 and plan.spawn_rate() > 0.0
+                 and plan.spawn_limit() > 0
+                 and plan.ship_designs())]
+    if not plans:
+        return
+
+    plan = plans[0]
+    spawn_limit = plan.spawn_limit()
+    placed = []
+    removed_lanes = set()
+
+    starlanes = [(s1, s2) for s1 in systems for s2 in fo.sys_get_starlanes(s1) if s1 < s2]
+
+    for system in systems:
+        # Quit when no more to spawn
+        if len(placed) >= spawn_limit:
+            return
```

Comment:
Implementing experimentor placement like that will cause different handling of the spawn limit parameter. The spawn limit is supposed to be the *max* limit, not the actual number of fleets to place. Unless I've reading your code wrong, this will cause `spawn_limit` experimentor monsters to be placed (or at least try to, provided that enough suitable systems are present).